interview-questions.md

```diff
@@ -1,61 +1,119 @@
 # ASSESSMENT 2: Interview Practice Questions
 
-Answer the following questions. First, without external resources. Challenge yourself to answer from memory. Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn.
+Answer the following questions. First, without external resources. Challenge
+yourself to answer from memory. Then, research the question to expand on your
+answer. Even if you feel you have answered the question completely on your own,
+there is always something more to learn.
 
 1. Why would you use the method super()?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: The super() method allows you to pass the variables and methods
+from the parent down to the child class component and allows the child class
+component to have the full functionality of the parent class.
 
+Researched answer: super() will call the constructor of its parent class.
+super() is called inside a react component only if it has a constructor. However
+if we have a constructor then super() is mandatory. This is required when you
+need to access some variables from the parent class. In React, when you call
+super with props. React will make props available across the component through
+this.props
 
 2. What is a virtual DOM?
 
-  Your answer:
-
-  Researched answer:
-
-
-
-3. When creating a basic (stateless) class component in React, what are the necessary elements needed to display "Hello World" in the browser?
+Your answer: The virtual DOM allows a React app to update components in real
+time without having to refresh the browser. The updates are affected by
+behavior/actions of the user on a component that change the state of a component
+in real time. (ie. clicking a button, typing into a search box or form)
 
-  Your answer:
+Researched answer: The virtual DOM (VDOM) is a programming concept where an
+ideal, or “virtual”, representation of a UI is kept in memory and synced with
+the “real” DOM by a library such as ReactDOM. This process is called
+reconciliation.
 
-  Researched answer:
+3. When creating a basic (stateless) class component in React, what are the
+   necessary elements needed to display "Hello World" in the browser?
 
+Your answer: The render method is not necessary. Only the return is necessary,
```

Comment:
Interesting about the render method not being necessary. Since the question is revolving around a **Class Based Component**, how about the class call, imports and exports?

Reply:
Yes, I misread the question. I was referencing a functional component.